float64 x
float64 y
---
int16 min_x
int16 max_x
int16 min_y
int16 max_y
bool safe
